test/basic.test.js

```diff
@@ -725,3 +725,13 @@ test('throws if prettyPrint is passed in as an option', async (t) => {
     })
   }, new Error('prettyPrint option is no longer supported, see the pino-pretty package (https://github.com/pinojs/pino-pretty)'))
 })
+
+test('Should invoke `onChild` with the newly created child', async ({ equal }) => {
+  let innerChild
+  const child = pino({
+    onChild: (instance) => {
+      innerChild = instance
+    }
+  }).child({ foo: 'bar' })
+  equal(child, innerChild)
```

Comment:
How do we know that `innerChild` has been set when this assertion is invoked? Shouldn't the assertion be within the callback and a `t.plan` added to the start of the subtest?

Reply:
The `onChild` function runs [here](https://github.com/pinojs/pino/pull/1541/files#diff-78c851c106059931b9042837eb4d521a7650b32535f306d9cb85bbddef22f877R144), right before the `child` instance is retuned by the `child` function, so that's how we could know that it's been set.
Over then that we couldn't make the assertion inside the callback since it will only have access to the `instance` and not the returned value `child`.
Please let me know if you still think any changes should be made.

Reply:
Are we asserting that the callback must be synchronous? If so, the documentation should be updated to make that clear.

Reply:
You are right, I've updated the documentation.